server/middleware/per-request.js

```diff
@@ -31,6 +31,16 @@ function perRequestContextFactory(options) {
   var scope = options.name || name;
   var enableHttpContext = options.enableHttpContext || false;
   var ns = LoopBackContext.createContext(scope);
+  // Monkey-patch LoopBackContext.getCurrentContext in order to fix issue #17
```

Comment:
I am wondering - why are we applying this workaround only in the middleware, instead of binding get/set methods directly inside the original `getCurrentContext`?

Also IIUC, this code is fixating the `context` object to a single value when a request reaches `per-request` middleware. How is this going to support the case when there are multiple requests handled in parallel?

Imagine the following handler chain (three different middleware handlers):

 1. per-context
 2. access-token retrieval, finishes in a later tick
 3. store access token in the context

Now consider two requests handled in the following sequence:

 1. Request1 invokes per-context, modifies global `LoopBackContext.getCurrentContext` to return CTX1
 2. Request1 starts access-token retrieval
 3. Request2 invokes per-context, modifies global `LoopBackContext.getCurrentContext` to return CTX2
 4. Request2 starts access-token retrieval
 5. Request1 access token is retrieved
 6. The access token is stored in CTX2. Ooops.